From: Richard Acayan <mailingradian@gmail.com>
Date: Tue, 25 Jun 2024 18:43:31 -0400
Subject: [PATCH] ASoC: qdsp6: q6afe: pass tdm ctrl_sync_data_delay from
 config

The codec may expect to delay data by 0, 1 or 2 bit clock cycles.
Keeping the TDM ctrl_sync_data_delay value zero-initialized could result
in corruption of data being read at the AFE. Add a configuration value
for the data delay and send it when preparing the TDM port.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 sound/soc/qcom/qdsp6/q6afe.c | 1 +
 sound/soc/qcom/qdsp6/q6afe.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sound/soc/qcom/qdsp6/q6afe.c b/sound/soc/qcom/qdsp6/q6afe.c
index 8b50c67..2d479fe 100644
--- a/sound/soc/qcom/qdsp6/q6afe.c
+++ b/sound/soc/qcom/qdsp6/q6afe.c
@@ -1316,6 +1316,7 @@ void q6afe_tdm_port_prepare(struct q6afe_port *port,
 	pcfg->tdm_cfg.sync_mode = cfg->sync_mode;
 	pcfg->tdm_cfg.sync_src = cfg->sync_src;
 	pcfg->tdm_cfg.nslots_per_frame = cfg->nslots_per_frame;
+	pcfg->tdm_cfg.ctrl_sync_data_delay = cfg->data_delay;
 
 	pcfg->tdm_cfg.slot_width = cfg->slot_width;
 	pcfg->tdm_cfg.slot_mask = cfg->slot_mask;
diff --git a/sound/soc/qcom/qdsp6/q6afe.h b/sound/soc/qcom/qdsp6/q6afe.h
index acd071d..72e9e2c 100644
--- a/sound/soc/qcom/qdsp6/q6afe.h
+++ b/sound/soc/qcom/qdsp6/q6afe.h
@@ -189,6 +189,7 @@ struct q6afe_tdm_cfg {
 	u16	sync_mode;
 	u16	sync_src;
 	u16	nslots_per_frame;
+	u16	data_delay;
 	u16	slot_width;
 	u16	slot_mask;
 	u32	data_align_type;
